packages/react-components/src/components/VideoGallery.tsx

```diff
@@ -171,3 +147,65 @@ export const VideoGallery = (props: VideoGalleryProps): JSX.Element => {
     </GridLayout>
   );
 };
+
+// Use React.memo to create memoize cache for each RemoteVideoTile
+const RemoteVideoTile = React.memo(
+  (props: {
+    userId: string;
+    onCreateRemoteStreamView?: (userId: string, options?: VideoStreamOptions | undefined) => Promise<void>;
+    onDisposeRemoteStreamView?: (userId: string) => Promise<void>;
+    isAvailable?: boolean;
+    renderElement?: HTMLElement;
+    displayName?: string;
+    remoteVideoViewOption?: VideoStreamOptions;
```

Comment:
It's a bit scary if a pure component takes structured props.

It's easy to change one of the fields inside `VideoStreamOptions` and incorrectly expect React to rerender.

Reply:
Thanks for looking into this PR deeply!
I agree that we need to keep the props minimum, for this case looks like every value in props is joining the life cycle (imagine that someone passes in a new option object, but the RemoteVideoTile content does not get updated. Or renderElement get a new one but it doesn't get hooked up to the inner content)

Maybe we can have more discussions on life cycle and design a better one for VideoTile, I will keep this PR focusing on fixing overuse of memoizeFnAll

Reply:
Can you create a work item in the backlog so this isn't forgotten about?